go.mod

```diff
@@ -1,20 +1,85 @@
 module github.com/open-cluster-management/discovery
 
-go 1.16
+go 1.17
 
 require (
-	github.com/gin-gonic/gin v1.7.2
+	github.com/gin-gonic/gin v1.7.4
 	github.com/go-logr/logr v0.4.0
 	github.com/onsi/ginkgo v1.16.4
-	github.com/onsi/gomega v1.13.0
+	github.com/onsi/gomega v1.16.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
-	github.com/rs/zerolog v1.20.0
+	github.com/rs/zerolog v1.25.0
 	github.com/stretchr/testify v1.7.0
-	k8s.io/api v0.21.1
-	k8s.io/apimachinery v0.21.1
-	k8s.io/client-go v0.21.1
-	k8s.io/metrics v0.21.1
-	sigs.k8s.io/controller-runtime v0.9.0
+	k8s.io/api v0.22.1
+	k8s.io/apimachinery v0.22.1
+	k8s.io/client-go v0.22.1
+	k8s.io/metrics v0.22.1
+	sigs.k8s.io/controller-runtime v0.10.0
 	sigs.k8s.io/yaml v1.2.0
 )
+
+require (
```

Comment:
Are all these needed? I see they are all //indirect and now we have 4x as many imports here

Reply:
It's a Go 1.17 change to include transitively-imported packages https://golang.org/doc/go1.17#go-command